edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/storage/MessageStore.cs

```diff
@@ -261,28 +261,38 @@ private async Task CleanQueue(bool checkEntireQueueOnCleanup)
                 {
                     foreach (KeyValuePair<string, ISequentialStore<MessageRef>> endpointSequentialStore in this.messageStore.endpointSequentialStores)
                     {
+                        var messageQueueId = endpointSequentialStore.Key;
                         try
                         {
                             if (this.cancellationTokenSource.IsCancellationRequested)
                             {
                                 return;
                             }
 
-                            Events.CleanupTaskStarted(endpointSequentialStore.Key);
-                            CheckpointData checkpointData = await this.messageStore.checkpointStore.GetCheckpointDataAsync(endpointSequentialStore.Key, CancellationToken.None);
+                            var (endpointId, priority) = MessageQueueIdHelper.ParseMessageQueueId(messageQueueId);
+                            Events.CleanupTaskStarted(messageQueueId);
+                            CheckpointData checkpointData = await this.messageStore.checkpointStore.GetCheckpointDataAsync(messageQueueId, CancellationToken.None);
                             ISequentialStore<MessageRef> sequentialStore = endpointSequentialStore.Value;
-                            Events.CleanupCheckpointState(endpointSequentialStore.Key, checkpointData);
+                            Events.CleanupCheckpointState(messageQueueId, checkpointData);
                             int cleanupEntityStoreCount = 0;
 
+                            // If cleanup just peek head, message counts is tailOffset-headOffset+1
```

Comment:
It would be clearer for me if the comment was
"If checkEntireQueueOnCleanup is set to false, we only peek the head, so message count is tailOffset-headOffset+1"